CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/TokenKindFile.swift

```diff
@@ -0,0 +1,393 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+let tokenKindFile = SourceFile {
+  EnumDecl("""
+    /// Enumerates the kinds of tokens in the Swift language.
+    @frozen // FIXME: Not actually stable, works around a miscompile
+    public enum TokenKind: Hashable
+    """) {
+    EnumCaseDecl("case eof")
+    
+    for token in SYNTAX_TOKENS {
+      // Tokens that don't have a set text have an associated value that
+      // contains their text.
+      if token.text == nil {
+        EnumCaseDecl("case \(raw: token.swiftKind)(String)")
+      } else {
+        EnumCaseDecl("case \(raw: token.swiftKind)")
+      }
+    }
+    
+    InitializerDecl("""
+      /// Initializes a keyword token kind from its string representation. If the
+      /// given string is not a keyword, this function returns `nil`.
+      public init?(keyword: String)
+      """) {
+      SwitchStmt(expression: Expr("keyword")) {
+        for token in SYNTAX_TOKENS where token.isKeyword {
+          SwitchCase("case \"\(raw: token.text!)\":") {
+            SequenceExpr("self = .\(raw: token.swiftKind)")
+          }
+        }
+        
+        SwitchCase("default:") {
+          ReturnStmt("return nil")
+        }
+      }
+    }
+    
+    VariableDeclSyntax(
+      leadingTrivia: .docBlockComment("/// The textual representation of this token kind.") + .newlines(1),
+      attributes: [AttributeListSyntax.Element.attribute(AttributeSyntax(attributeName: .identifier("_spi(Testing)\n")))],
+      modifiers: [DeclModifier(name: .public)],
+      name: IdentifierPattern("text"),
+      type: TypeAnnotation(type: TypeSyntax("String"))
+    ) {
+      SwitchStmt(expression: ExprSyntax("self")) {
+        for token in SYNTAX_TOKENS {
+          
+          if let text = token.text {
+            SwitchCase("case .\(raw: token.swiftKind):") {
+              ReturnStmt("return #\"\(raw: text)\"#")
+            }
+          } else {
+            SwitchCase("case .\(raw: token.swiftKind)(let text):") {
+              ReturnStmt("return text")
+            }
+          }
+        }
+        
+        SwitchCase("case .eof:") {
+          ReturnStmt(#"return """#)
+        }
+      }
+    }
+    
+    VariableDeclSyntax(
+      leadingTrivia: .docBlockComment("""
+        /// Returns `true` if the token is a Swift keyword.
```

Comment:
I think each line needs to be its own `docBlockCommen` trivia so this is indented correctly.